api/v1alpha3/awsmachinetemplate_webhook.go

```diff
@@ -60,7 +62,7 @@ func (r *AWSMachineTemplate) ValidateCreate() error {
 func (r *AWSMachineTemplate) ValidateUpdate(old runtime.Object) error {
 	oldAWSMachineTemplate := old.(*AWSMachineTemplate)
 	if !reflect.DeepEqual(r.Spec, oldAWSMachineTemplate.Spec) {
-		return errors.New("awsMachineTemplateSpec is immutable")
+		return errImmutableSpec
```

Comment:
Either here or in a follow-up could you make it match the convention seen at https://github.com/kubernetes-sigs/cluster-api/blob/38d49ae7e8c328790ad74e935b5af207d75b7064/api/v1alpha3/machine_webhook.go#L132-L135 and use something like this to indicate the field is immutable?

```go
field.Invalid(field.NewPath("spec", "clusterName"), m.Spec.ClusterName, "field is immutable")
```

Reply:
I actually did that at first, but it results in a really nasty error message that spews out the entire spec back to the user.

Reply:
Hmm, ok. How about returning `apierrors.NewInvalid()` with the text about spec being immutable?

Reply:
You mean like

```
return apierrors.NewInvalid(GroupVersion.WithKind("AWSMachineTemplate").GroupKind(), r.Name, field.ErrorList{
    field.Invalid(field.NewPath("spec"), r.Spec, "is immutable"),
})
```

This has the same problem, the error message is 

```
error: awsmachinetemplates.infrastructure.cluster.x-k8s.io "bmo-control-plane" could not be patched: admission webhook "validation.awsmachinetemplate.infrastructure.x-k8s.io" denied the request: AWSMachineTemplate.infrastructure.cluster.x-k8s.io "bmo-control-plane" is invalid: spec: Invalid value: v1alpha3.AWSMachineTemplateSpec{Template:v1alpha3.AWSMachineTemplateResource{Spec:v1alpha3.AWSMachineSpec{ProviderID:(*string)(nil), AMI:v1alpha3.AWSResourceReference{ID:(*string)(nil), ARN:(*string)(nil), Filters:[]v1alpha3.Filter(nil)}, ImageLookupFormat:"", ImageLookupOrg:"", ImageLookupBaseOS:"", InstanceType:"t3.larges", AdditionalTags:v1alpha3.Tags(nil), IAMInstanceProfile:"control-plane.cluster-api-provider-aws.sigs.k8s.io", PublicIP:(*bool)(nil), AdditionalSecurityGroups:[]v1alpha3.AWSResourceReference(nil), FailureDomain:(*string)(nil), Subnet:(*v1alpha3.AWSResourceReference)(nil), SSHKeyName:(*string)(0xc0002baa60), RootVolume:(*v1alpha3.RootVolume)(nil), NetworkInterfaces:[]string(nil), UncompressedUserData:(*bool)(nil), CloudInit:v1alpha3.CloudInit{InsecureSkipSecretsManager:false, SecretCount:0, SecretPrefix:""}}}}: is immutable
```

Reply:
Try it with field.Forbidden instead of Invalid (assuming you agree that Forbidden is acceptable here).

Reply:
`return apierrors.NewBadRequest("AWSMachineTemplate.Spec is immutable")` seems to do the trick:

> error: awsmachinetemplates.infrastructure.cluster.x-k8s.io "bmo-control-plane" could not be patched: admission webhook "validation.awsmachinetemplate.infrastructure.x-k8s.io" denied the request: AWSMachineTemplate.Spec is immutable

Reply:
:shipit: